compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/symbols/BallerinaRecordTypeSymbol.java

```diff
@@ -114,9 +114,12 @@ public String signature() {
             joiner.add(ballerinaFieldSignature);
         }
 
-        restTypeDescriptor().ifPresent(typeDescriptor -> {
-            joiner.add(typeDescriptor.signature() + "...;");
-        });
+        // No rest descriptor if the record in inclusive
+        if (!this.isInclusive) {
```

Comment:
Do you need this inclusivity info? The issue with using this is this check is not accurate. We don't keep this info in the internal symbol, so it's not available here either.

Reply:
The issue is because of #28695 
And I'm not sure why it cannot be used, the same check is used in L107 few lines above this.
https://github.com/ballerina-platform/ballerina-lang/blob/b235daf9bc6576b5a015ca256751efbec98023bb/compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/symbols/BallerinaRecordTypeSymbol.java#L107-L109

Reply:
The issue is this check doesn't always return the correct result. For example, this would incorrectly evaluate to true for the following:
```ballerina
type Person record {|
    string name;
    int age;
    string...;
|};
```

Reply:
This is something we need to fix in the compiler to get this check to correctly work. At the moment, this syntactic difference between inclusive and exclusive record typedescs is not captured by internal type info since it has no bearing on the compiler analysis phases.

Reply:
I changed the signature to treat the symbol as inclusive iif `this.isInclusive` is true and the rest type (if there is any)  is `anydata`.

Reply:
That still wouldn't be accurate right? Suppose the user defines a record as follows: 
```ballerina
type Person record {|
    string name;
    int age;
    anydata...;
|};
```
Now `Person` would be incorrectly identified as an inclusive record typedesc.

Reply:
Isn't,
```ballerina
type Person record {|
    string name;
    int age;
    anydata...;
|};
```

same as 
```ballerina
type Person record {
    string name;
    int age;
};
```

so cant we treat closed records with anydata field as the same as open records.